src/module.c

```diff
@@ -6224,6 +6230,7 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
                 sds msg = sdsnew("cannot run as user, no user directly attached to context or context's client");
                 reply = callReplyCreateError(msg, ctx);
             }
+            moduleFreeArgv(argv, argc);
```

Comment:
maybe a better fix would be to move this check to after we're allocating `c` and setting it's `c->argv`, and then do `goto cleanup` instead of returning?
this way the error is also in `autoMemoryAdd`.
it would mean some unnecessary overhead in this path, but it's not a path that we really expect to ever happen on a module that's properly written, right?

p.s. it should also be below the line that does `in_call++;`.

Reply:
@sjpotter please have a look at my change (using the fix i described above instead of yours)

Reply:
ohh, now i see the problem.
also also learn (again) not to edit code in GH browser where i can't validate it works before committing.

Reply:
based on your fix, I'm not sure it makes sense for the `moduleAllocTempClient()` function to take a user struct anymore (the reason I like it taking the user struct as there's symmetry with the release function).  But this would fix it as well, so either way is fine to me.

Reply:
ohh, right, this was the only place that passed a non-NULL value.
let's drop that argument.

Reply:
reverted to the state before 6e993a5dfa

i'm a bit disappointed that this PR resulted in so many changes (would make it harder to backport).
but on the other hand, i'm quite confident that the right approach was to move this setup block to a place where en error with `goto cleanup` is possible.